src/kernel-install/kernel-install

```diff
@@ -155,8 +155,17 @@ done
 [ -z "$ENTRY_TOKEN" ] && ENTRY_TOKEN="$MACHINE_ID"
 
 if [ -z "$layout" ]; then
-    # Administrative decision: if not present, some scripts generate into /boot.
-    if [ -d "$BOOT_ROOT/$ENTRY_TOKEN" ]; then
+    # No layout configured by the administrator. Let's try to figure it out automatically
+    if [ -e "$BOOT_ROOT/loader/entries.srel" ]; then
+        read -r ENTRIES_SREL <"$BOOT_ROOT/loader/entries.srel"
+        if [ "$ENTRIES_SREL" == "standard" ]; then
+            # The loader/entries.srel file clearly indicates that this uses the proper standard boot loader spec
+            layout="bls"
+        else
+            layout="other"
```

Comment:
Shoudn't we assume "layout=xxx" when "xxx" is specified in `entries.srel`? If you use some non-standard entry format, it seems reasonable to assume that you're using a corresponding layout of entries.

I'm not sure if it makes sense to create a separate file for this. We already have `layout.conf`: why not add a field there instead? We already put random-seed config in loader.conf.

Reply:
this new file is placed in the ESP/XBOOLDR partition, not /etc/kernel/. I.e. it's about describing which format the boot loader installed in the ESP expects its /loader/entries files. And that's different from selecting in which format kernel-install *creates* these entries (whch is stored in /etc/kernel/install.conf, and for which there already is `$layout` field there).

Hence, this new file is purely informational: it reports "hey, we have a boot loader installed where the /loader/entries/ dir expects standards compliant entries". That makes it quite different from the existing /etc/kernel/install.conf entry which is not informational, but is admin-config: it tells kernel-install "hey you, when you write to /loader/entries/, use this format".

Now, the two fields are not entirely unrelated, after all if there's already a boot loader installed, we should generally install entries in the format it understands. But it's also quite different:

1. One day, once kernel-install learns generating type `#2` boot loader spec entries it might make sense to write `echo layout=unified >> /etc/kernel/install.conf` to select that it should create entries in that format now. But  `/loader/entries.srel` should never carry a string such as "unified", because it is purely about the format of entries in the `/loader/entries/` directory

2. if no boot loader spec is installed yet, but kernel-install invoked we require explicit configuration in which format to generate the entries.

Reply:
And to reply more explicitly:

> Shoudn't we assume "layout=xxx" when "xxx" is specified in entries.srel? If you use some non-standard entry format, it seems reasonable to assume that you're using a corresponding layout of entries.

If my boot loader supports both our type 1 and type 2 entry types (which is exacly what sd-boot does) then `$ESP/loader/entries.srel` should contain `standard` (as that#s hte format `$ESP/loader/entries/` drop-ins should have. But I might still want to create files in type 2 format instead, hence such an implicaiton looks wrong. Moreover, I think the grub implementation of our spec might actually support our dropins but not vice versa. Hence the choice of what to generate is not that clear: we might even want to generate our format in case the grub format is supported.

> I'm not sure if it makes sense to create a separate file for this. We already have layout.conf: why not add a field there instead? We already put random-seed config in loader.conf.

loader.conf is an sd-boot config file, and not part of the boot loader spec. Do you mean /etc/kernel/install.conf when you say `layout.conf` (not aware of any file by that name). If so, it's nt in the ESP, so doens't work.

Reply:
Yeah, I meant `loader.conf`, sorry for the typo.

OK, I guess it's reasonable to add a new file.